### PR TITLE
fix: link escaping in booking page

### LINKED ISSUE #10352: Link meeting omits 'https:' on booking page when 'Display on booking page' is chosen
### Steps to Reproduce

1. Create an event with a meeting location of `Link Meeting` make sure the `display on booking page checkbox is checked`
Use this link
https://umd.zoom.us/j/6279782951?pwd=changemeplease

image

2. Visit the booking page
image

The link appears as `//umd.zoom.us/j/6279782951?pwd=changemeplease` with the protocol omitted

### Expected Results

I would expect the full link to be shown on the booking page

### Technical details
Brave 1.56.11

### Thoughts
I think this is an HTML sanitizer scrubbing the links in the UI. If this is the case, the rules should be relaxed for this field to show whatever URL the author chooses.

Of course, an ftp:// wouldn't make sense, but https is a must.

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

Fixes #10352

## Requirement/Documentation

before:
Screenshot 2023-07-25 at 14-31-42 test Admin Example Cal com

after:

Screenshot 2023-07-25 at 14-30-35 test Admin Example Cal com

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/features/bookings/lib/getLocationOptionsForSelect.ts [hunk 1]
```diff
@@ -19,10 +19,17 @@ export default function getLocationsOptionsForSelect(
...
 
+      // to prevent escaping of "http" when the link is displayed in the booking page
+      const shouldLinkDisplayedPublicly = type === "link" && locationString !== "link_meeting";
+
       return {
-        label: t(locationString),
+        label: shouldLinkDisplayedPublicly ? locationString : t(locationString),
         value: type,
-        inputPlaceholder: t(eventLocation?.attendeeInputPlaceholder || ""),
+        inputPlaceholder: t(eventLocation?.attendeeInputPlaceholder || "", {
```
Comment (hariombalhara) [inline packages/features/bookings/lib/getLocationOptionsForSelect.ts:28]: @G3root Does` t` fn remove http: ?

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thanks @G3root 🙏 Left some comments

**Comment by G3root** [comment]
@hariombalhara pushed some changes to make it typesafe. tried `interpolation: {escapeValue:false}` but it didn't work out

**Review by hariombalhara — APPROVED** [review]
Fixed the root cause @G3root . Left a comment in the code as it was super weird.

**Pull request merged**